api/health/worker.go

```diff
@@ -12,41 +12,71 @@ import (
 
 	"github.com/prometheus/client_golang/prometheus"
 
+	"go.uber.org/zap"
+
 	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
 
 	"github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/utils/logging"
 	"github.com/ava-labs/avalanchego/utils/set"
 )
 
-var errDuplicateCheck = errors.New("duplicated check")
+var (
+	allTags = []string{AllTag}
+
+	errRestrictedTag  = errors.New("restricted tag")
+	errDuplicateCheck = errors.New("duplicated check")
+)
 
 type worker struct {
+	log        logging.Logger
+	namespace  string
 	metrics    *metrics
 	checksLock sync.RWMutex
-	checks     map[string]Checker
+	checks     map[string]*checker
 
-	resultsLock sync.RWMutex
-	results     map[string]Result
-	tags        map[string]set.Set[string] // tag -> set of check names
+	resultsLock                 sync.RWMutex
+	results                     map[string]Result
+	numFailingApplicationChecks int
+	tags                        map[string]set.Set[string] // tag -> set of check names
 
 	startOnce sync.Once
 	closeOnce sync.Once
 	wg        sync.WaitGroup
 	closer    chan struct{}
 }
 
-func newWorker(namespace string, registerer prometheus.Registerer) (*worker, error) {
+type checker struct {
```

Comment:
I'm not super happy about having a `checker` struct that has a `checker Checker` field so if you've got a better naming idea I'd be very amenable lol

Reply:
Maybe this could be like, `taggedChecker` or something?

Reply:
sgtm